### PR TITLE
ignore nil error on WithError (fixes #67)

### LINKED ISSUE #67: Ignore nil errors in WithError
Would you accept a PR that adds 

```go
if err == nil {
    return e
}
```

To [this line](https://github.com/apex/log/blob/941dea75d3ebfbdd905a5d8b7b232965c5e5c684/entry.go#L55)? `pkg/errors` and other log implementations have this functionality as well.

### PR DESCRIPTION
**Pull request by blockloop**

### CODE REVIEW

**Comment by blockloop** [comment]
Ah okay. We have some pieces of configuration where things flow through certain areas of code if there is an error or not and often we have the same log message and fields regardless. Something like 

```go
n, err := strconv.Atoi(num)
log.WithField("export_level", n).
		WithError(err).
		Info("exporter level configured")
```

We want to log regardless of the error so it requires a bit more code to accomplish

```go
n, err := strconv.Atoi(num)
ll := log.WithFields("export_level", n)
if err != nil {
    ll = ll.WithError(err)
}
ll.Info("exporter level configured")
```

**Comment by tj** [comment]
ahhh I guess that's a little unconventional as well since it's not an `Error("xxx")`. Might be nice to have some kind of conditional error/info method which handles the nil error, not sure how I feel about it haha

**Comment by jackmcguire1** [comment]
+1 for this

File: entry.go [hunk 1]
```diff
@@ -52,6 +52,10 @@ func (e *Entry) WithField(key string, value interface{}) *Entry {
 // The given error may implement .Fielder, if it does the method
 // will add all its `.Fields()` into the returned entry.
 func (e *Entry) WithError(err error) *Entry {
+	if e == nil {
```
Comment (timbunce) [inline entry.go]: Shouldn't this be `if err == nil {` or am I misunderstanding the intent?

Comment (tych0) [reply entry.go:55]: Agreed, it should be.

**Comment by tj** [comment]
@jackmcguire1 unfortunately Go doesn't agree :D, otherwise it wouldn't have `nil` to begin with

**Comment by jackmcguire1** [comment]
@tj apologies but I’m sure none of us are expecting a nil panic, much like why this thread was created in the first place, it’s not at all necessary to panic, especially in the scenario if we don’t want to cause unnecessary nesting if statements...

I’m just better off doing

‘’’
log.WithField(“err”, err) to be safe....

**Pull request closed without merging**